src/utils/queries.rs

```diff
@@ -9,197 +9,453 @@ use super::{
         NEUTRON_CHAIN_NAME, OSMOSIS_CHAIN_NAME, PAIR_NAME, PRICE_ORACLE_NAME, STABLE_PAIR_NAME,
         TX_HASH_QUERY_PAUSE_SEC, TX_HASH_QUERY_RETRIES,
     },
-    test_context::TestContext,
+    test_context::{LocalChain, TestContext},
 };
-use localic_std::modules::cosmwasm::CosmWasm;
+use localic_std::{modules::cosmwasm::CosmWasm, transactions::ChainRequestBuilder};
 use serde_json::Value;
 use std::{path::PathBuf, thread, time::Duration};
 
-impl TestContext {
-    /// Gets the event log of a transaction as a JSON object,
-    /// or returns an error if it does not exist.
-    pub fn guard_tx_errors(&self, chain_name: &str, hash: &str) -> Result<(), Error> {
-        if !self.unwrap_logs {
-            return Ok(());
+pub enum QueryType {
```

Comment:
Wondering if it might be worth splitting this enum into different QueryTypes inside.

Something like IBCQueries, CosmwasmQueries, OsmosisQueries, AstroportQueries, ValenceQueries, feels like it would be cleaner.

Reply:
This way the matches you do in the get functions might look a bit shorter and more organized. Not a blocker tho